File: frontend/src/services/statsService.js
import apiClient from './api';

export default {
  // Get all statistics
  getAllStats() {
    return apiClient.get('/stats');
  },

  // Get overview statistics
  getOverview() {
    return apiClient.get('/stats/overview');
  },

    // 获取时间段统计数据
    getTimelineStats(params) {
      return apiClient.get('/stats/timeline', { params });
    },

  // Get category statistics
  getCategoryStats() {
    return apiClient.get('/stats/categories');
  },

  // Get wrong questions statistics
  getWrongQuestions() {
    return apiClient.get('/stats/wrong-questions');
  },

  // Get session records
  getSessions() {
    return apiClient.get('/stats/sessions');
  },

  // Get learning trend data
  getTrends() {
    return apiClient.get('/stats/trends');
  },

  // Start a new study session
  startSession(mode = 'normal') {
    return apiClient.post('/stats/sessions/start', { mode });
  },

  // End current study session
  endSession() {
    return apiClient.post('/stats/sessions/end');
  },

  // Reset all statistics
  resetStats() {
    return apiClient.post('/stats/reset');
  },

  // Get advisor suggestions
  getAdvisorSuggestions() {
    return apiClient.get('/stats/advisor');
  },

  // Update view history
  updateViewHistory(view, currentIndex) {
    return apiClient.post('/stats/view-history', { view, currentIndex });
  },

  // Get bank progress
  getBankProgress() {
    return apiClient.get('/stats/bank-progress');
  },

  // Get tag progress
  getTagProgress() {
    return apiClient.get('/stats/tag-progress');
  },

  // Get mastery overview
  getMastery() {
    return apiClient.get('/stats/mastery');
  }
};

File: backend-springboot/src/main/java/com/quizgen/app/stats/service/StatsService.java
package com.quizgen.app.stats.service;

import com.quizgen.app.study.entity.StudySession;

import java.util.List;
import java.util.Map;

public interface StatsService {

    StudySession startStudySession(String mode);

    long endStudySession();

    void recordAnswer(Long questionId, boolean isCorrect);

    void addWrongQuestion(Long questionId);

    void removeWrongQuestion(Long questionId);

    List<Long> getWrongQuestionIds();

    void updateViewHistory(String view, int currentIndex);

    Map<String, Object> getStatsRaw();

    Map<String, Object> getOverview();

    Map<String, Object> getCategoryStats();

    Map<String, Object> getWrongQuestions();

    List<Map<String, Object>> getSessions();

    Map<String, Object> getTrends();

    Map<String, Object> getAdvisorSuggestions();

    Map<String, Object> getTimeline(String period, String startDate, String endDate, String category);

    List<Map<String, Object>> getBankProgress();

    List<Map<String, Object>> getTagProgress();

    Map<String, Object> getMasteryOverview();

    void resetStats();
}